plugins/hls-pragmas-plugin/src/Ide/Plugin/Pragmas.hs

```diff
@@ -187,7 +188,19 @@ completion _ide _ complParams = do
 -- Useful for inserting pragmas.
 endOfModuleHeader :: T.Text -> Range
 endOfModuleHeader contents = Range loc loc
-    where
-        loc = Position line 0
-        line = maybe 0 succ (lastLineWithPrefix "{-#" <|> lastLineWithPrefix "#!")
-        lastLineWithPrefix pre = listToMaybe $ reverse $ findIndices (T.isPrefixOf pre) $ T.lines contents
+  where
+    loc = Position line 0
+    line = maybe 0 succ (lastShebang contents <|> lastOptsGhc contents <|> lastLangPrag contents)
+    lastShebang = lastLineWithPrefix $ T.isPrefixOf "#!"
```

Comment:
Could you expand the doc comment (and fix the markup) to explain what the new behaviour is? It will make it easier to review and to understand in the future.